HW 3 Completed

Kesmail03 opened the pull request:
**01-Coins.js**
For this file I first tried to keep the amount the same and then trying to do the operations but I couldn't get it to work. I believe it has something to do with javascript floating point numbers so I changed the dollar amount to cents and got it to work from there using a while loop. I thought about using a switch case state but the while loop worked better

**02-node-routing.js**
For this file I just used the if and else if statements depending on the route. I did have to use the dev tools to verify that the cookies and caching was working properly. To do so I went to the network section to check the cache and storage section to see if the cookie was set. 

**03-express-routing.js**
Surprisingly I found that the syntax for express.js is quite similar to the flask framework in python which I found interesting. I did have to review the different HTTP methods to make sure I was using GET and POST correctly because you have to use the right method with app. This one was very similar to the last one so it wasn't difficult.

**04-form.js**
This one was quite difficult at first. There are different ways to accept user input but I decided to use body parser. Here I was able to write plain html code to write a form using the get method. Then when the form was submitted I would use the post method to save the data I received earlier and output it to the user. It was difficult deciding which method I wanted to use at first but it was simple towards the end. 

I didn't use many test cases in the first file but I hope I tested everything correctly and it all works. I'm curious to see how everyone did the first file. Let me know if my code is wrong or if you think I could've done things more efficiently.

AndyDuong12 commented:
Hi Khaled,
### 01-Coins.js
- For this one, I think your codes look great. I have some suggestions I think that might be good to change but not necessary:
- When checking whether the user inputs higher than $100, you can check it before multiplying the change with 100, so it is easier to read the code. Seeing **10000** can be confusing where this number comes from. You can directly compare using `if(change > 100)`
- You can also add more edge cases: when the user inputs negative number, when the dollar is 0, etc.

### 02-node-routing.js
- This one looks good! I don't think there is anything that needs to be changed.

### 03-express-routing.js
- This exercise also looks good with no changes needed!

### 04-form.js
- This exercise also looks solid! I didn't think of putting all the html into one `res.send` which makes it look cleaner.

gaelg311 commented:
01- Code looks good, like Andy mentioned one thing I would change is to put the check for the maximum value at the top of the function so there isn't any unnecessary initialization of the coin values. Other than that the tests ran well! 
02- Exercise looks and works well, no changes needed 
03- Same as above, no changes needed 
04- The form looks good, and submits as necessary. The only change I would make to the form is making the newsletter a radio input as opposed to a text box. Other than that it looks good!

Kesmail03 commented:
Thanks for the feedback. In my new code I updated the first file. I made sure to do the test cases before changing the total amount to the equivalent in cents. I also added a few more test cases to see if the amount is zero or a negative number.
